feat: add support for multiple database connections

davelsan opened the pull request:
## Issue

This PR addresses issue #143.

## Description

#### [GMSC Issue 143](https://hackmd.io/PrqnC0bOTkG2e-C8BInxgA?view)
This PR implements the code generator changes necessary to query arbitrary local database connections defined at the data models layer. It also synchronizes `graphql-server-model-codegen` with changes in branch [`graphql-server#143-multiple-databases`](https://github.com/Zendro-dev/graphql-server/commits/issue%23143-multiple-databases).

#### [Integration Tests Changes](https://hackmd.io/vroDn6I4RbuY0IOGGI8l5w?view)

Because there is now a static `models/adapters/index.js` file in the `graphql-server`, the previous method of mounting generated code folders into the Docker image is no longer possible.

This has led to a different strategy for setting server instances dynamically, in addition to some improvements and optimizations to the integration tests CLI. **The API remains mostly the same**, with the flags we know and have been using until now having the same functionality.

To run integration tests using a specific `graphql-server` branch, the `Dockerfile.graphql_server` image file no longer needs to be modified. In fact, the server repository is no longer cloned in the image, which now only provides environment and tooling. An additional `-b <branch>` flag has been added to that effect.

```bash
# To execute a default run using the relevant branch for this PR
npm run test-integration -b "issue#143-multiple-databases"

# To execute a "keep containers alive" run
npm run test-integration -b "issue#143-multiple-databases" -k

# To execute a "generate code and run tests" and "keep containers alive"
npm run test-integration -b "issue#143-multiple-databases" -T -k
```

Other flags have also been added for a more modular functionality:

```bash
# To generate code only (this no longer ups the containers, as it was redundant with the other compatible flags)
npm run test-integration -g

# To clean up the generated code only
npm run test-integration -C

# To perform an initial setup: clones the server and re-creates instances
npm run test-integration -s

# Alternatively, an initial setup is compatible with the -b flag
npm run test-integration -b "issue#143-multiple-databases" -s
```

See the attached diagram for a visual explanation of how the API works.
[fzj-zendro-integration-tests-cli.pdf](https://github.com/Zendro-dev/graphql-server-model-codegen/files/5131718/fzj-zendro-integration-tests-cli.pdf)

## Blocks

- None. But it should merged along with [`graphql-server#42`](https://github.com/Zendro-dev/graphql-server/pull/42).

## Notes

- There is an API change in the `graphql-server`. Depending on whether there is a version sync between the two repos, this package might have to be bumped to the next `minor` version.

coeit commented on docker/integration_test_run/scripts/clean-workspace.sh:
```diff
@@ -0,0 +1,41 @@
+    # Checkout deleted static files
+    git checkout $(git diff --no-renames --name-only --diff-filter=D)
+
+    # Return to ReconstructDb root directory
```
remove "ReconstructDb"

coeit commented on test/sh_integration_test_run.sh:
```diff
@@ -15,39 +15,77 @@
 #   Execution via npm:
 #
-#     npm run test-integration [-- OPTIONS]
+#     yarn test-integration [-- OPTIONS]
```
Like we agreed: yarn not mandatory yet, I think we will use it though...

coeit reviewed (approved):
Approved for merging

davelsan commented:
Fixed the issues we talked about.

Changes since last revision:

- Refactored multi-purpose functions (fine-grained control allows better logs and debugs)
- Standardized and improved logs
- GraphQL server checks are now async (reduced wait time to 60s).

davelsan commented:
Fixed remaining issues and bumped to `0.3.0`.

Should be ready to merge along with [`graphql-server#42-multiple-databases`](https://github.com/Zendro-dev/graphql-server/pull/42)
